src/lib/models/abstraction-period.js

```diff
@@ -53,6 +101,80 @@ class AbstractionPeriod extends Model {
     assertMonth(month);
     this._endMonth = month;
   }
+
+  /**
+   * Sets the abstraction period
+   *
+   * @param {Number} startDay
+   * @param {Number} startMonth
+   * @param {Number} endDay
+   * @param {Number} endMonth
+   */
+  setDates (startDay, startMonth, endDay, endMonth) {
+    this.startDay = startDay;
+    this.startMonth = startMonth;
+    this.endDay = endDay;
+    this.endMonth = endMonth;
+  }
+
+  /**
+   * Checks if the passed AbstractionPeriod contains this instance. In order
+   * for this to return true, this instance must fit inside the passed
+   * AbstractionPeriod, but not matching the bounaries.
```

Comment:
I'm fairly sure matching the boundaries should also be included.
E.g. abs period from 1/4 to 31/10 would count as summer etc.